src/main/java/org/springframework/metrics/instrument/prometheus/PrometheusMeterRegistry.java

```diff
@@ -132,12 +136,38 @@ public double get() {
             };
 
             gauge.setChild(child, labelValues);
-            return new PrometheusGauge(name, child);
+            return new PrometheusGauge(id, child);
         });
 
         return obj;
     }
 
+    @Override
+    public MeterRegistry register(Meter meter) {
+        Collector collector = new Collector() {
+            @Override
+            public List<MetricFamilySamples> collect() {
+                List<MetricFamilySamples.Sample> samples = stream(meter.measure().spliterator(), false)
+                        .map(m -> {
+                            List<String> tagKeys = new ArrayList<>(m.getTags().size());
+                            List<String> tagValues = new ArrayList<>(m.getTags().size());
+                            for (Tag tag : m.getTags()) {
+                                tagKeys.add(tag.getKey());
+                                tagValues.add(tag.getValue());
+                            }
+                            return new MetricFamilySamples.Sample(m.getName(), tagKeys, tagValues, m.getValue());
```

Comment:
> Is that really more of a MeterBinder in our language -- i.e. something registering multiple counters?

I like that actually. I'm used to Prometheus where both types are collectors, but I think using binders for integrations makes good sense and keeps integrators from having to dig into internals too much.